onc-healthit/lantern-back-end
# Lantern-208

Changes in this PR:
The capability querier now adds entries for endpoints whose GET requests have timed out. These entries look similar to how the fhir_endpoinit_info_entry looks when the server request results in an error, as most fields are set to null, the http_response is set to 0, and the errors field explains that the server could not be reached. Issues in which endpoints were being added to the fhir_endpoints_info table with different formatting, which was due to a problem with capabilityquerier converting go's url type to a string url, has also been addressed. Testing files have also been updated to reflect these changes.

TO TEST:
- Start up and run lantern project like you normally would and open database
- Wait for capability querier to finish running (seems to take a little while)
- The query "SELECT url FROM fhir_endpoints WHERE url NOT IN (SELECT url FROM fhir_endpoints_info)" should result in an empty table
- Previously urls like complexmotion.com, https://api.baptist-health.org/Interconnect-FHIR/api/FHIR/DSTU2/, and SSMheatlh.com were not being added to fhir_endpoints_info table- can do a query to check to see if they are put into the table with the correct http response code and error
- The URLs https://secure64.oncoemr.com/OncoEMR# and HTTP://mhcmswscxsf01.mclaren.org/citrix/mclarenweb/ were not showing up in fhir_endpoints_info table due to string formatting errors, can do a query to make sure they now are included in the table 

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: 
[JIRA Ticker URL](https://oncprojectracking.healthit.gov/support/browse/LANTERN-208)
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
